src/classic/element/ReactElementValidator.js

```diff
@@ -213,19 +199,21 @@ function validateChildKeys(node, parentType) {
   } else if (node) {
     var iteratorFn = getIteratorFn(node);
     // Entry iterators provide implicit keys.
-    if (iteratorFn && iteratorFn !== node.entries) {
-      var iterator = iteratorFn.call(node);
-      var step;
-      while (!(step = iterator.next()).done) {
-        if (ReactElement.isValidElement(step.value)) {
-          validateExplicitKey(step.value, parentType);
+    if (iteratorFn) {
```

Comment:
No, this is fixing that behavior. Note the next line that ignores validation of Maps. Previously it would try to validate the Map as if it was an object, which depending on the type of Map might fail.

(I'm not sure Maps (iterables of `[key, value]`) should be a supported way to key children. I think it makes more sense to turn Map data into iterables of ReactElement which is effectively a `[key, value]` tuple where key is `type + key`. But that is unrelated to this PR.)